codersdk/notifications.go

```diff
@@ -212,4 +212,72 @@ type UpdateNotificationTemplateMethod struct {
 
 type UpdateUserNotificationPreferences struct {
 	TemplateDisabledMap map[string]bool `json:"template_disabled_map"`
+	PushSubscription    string          `json:"push_subscription,omitempty"`
+}
+
+type WebpushMessageAction struct {
+	Label string `json:"label"`
+	URL   string `json:"url"`
+}
+
+type WebpushMessage struct {
+	Icon    string                 `json:"icon"`
```

Comment:
In the notification payload we use version for potential schema changes. Should we adopt this pattern too?

Reply:
Not a bad idea! It might be slightly premature right now, but I could see this being useful down the line.